src/core/structure/sheet.ts

```diff
@@ -435,46 +450,161 @@ export default class Sheet {
     return cellStyle;
   }
 
-  setCellStyle(
-    colKey: ColumnKey,
-    rowKey: RowKey,
-    style: CellStyle | null,
-  ): boolean {
-    const col = this.columns.get(colKey);
-    const row = this.rows.get(rowKey);
-    if (!col || !row) return false;
+  private getCellGroupByIndex(columnIndex: number, rowIndex: number) {
+    let columnKey = this.columnPositions.get(columnIndex);
+    let rowKey = this.rowPositions.get(rowIndex);
 
-    if (style == null) {
-      return this.clearCellStyle(colKey, rowKey);
+    if (!columnKey || !rowKey) {
+      const newCellElement = this.initializePosition(columnIndex, rowIndex);
+      columnKey = newCellElement.columnKey;
+      rowKey = newCellElement.rowKey;
     }
+    return {
+      column: this.columns.get(columnKey!),
+      row: this.rows.get(rowKey!),
+      columnKey,
+      rowKey,
+    };
+  }
 
-    // TODO Style could be non-null but empty; should we allow this?
-    style = new CellStyle().clone(style);
+  setCellFormatter(
+    columnIndex: number,
+    rowIndex: number,
+    formatter: Formatter | null = null,
+  ): void {
+    const { column, row, columnKey, rowKey } = this.getCellGroupByIndex(
+      columnIndex,
+      rowIndex,
+    );
 
-    col.cellFormatting.set(row.key, style);
-    row.cellFormatting.set(col.key, style);
+    if (!column || !row) return;
 
-    if (style.formatter) {
-      this.applyCellFormatter(this.getCell(colKey, rowKey)!, colKey, rowKey);
+    if (formatter == null) {
+      column.cellFormatting.get(row.key)?.clearFormatter();
+      row.cellFormatting.get(column.key)?.clearFormatter();
+    } else {
+      column.cellFormatting.set(
+        row.key,
+        new CellStyle(column.cellFormatting.get(rowKey!)?.styling, formatter),
+      );
+      row.cellFormatting.set(
+        column.key,
+        new CellStyle(row.cellFormatting.get(columnKey!)?.styling, formatter),
+      );
     }
 
-    return true;
+    const cell = this.getCell(columnKey!, rowKey!);
+    this.applyCellFormatter(cell!, columnKey!, rowKey!);
```

Comment:
`cell` can be null here I think.

Reply:
I dont think so cz we are using initializePosition at the start of the method.

Reply:
`initializePosition` does not create a Cell object.

Reply:
what should i do here then?